core/src/main/java/org/elasticsearch/index/query/PrefixQueryParser.java

```diff
@@ -74,7 +76,7 @@ public PrefixQueryBuilder fromXContent(QueryParseContext parseContext) throws IO
                     }
                 }
             } else {
-                if (parseContext.parseFieldMatcher().match(currentFieldName, NAME_FIELD)) {
+                if (parseContext.parseFieldMatcher().match(currentFieldName, NAME_FIELD)) {   // this looks weird given that we already parse the _name field above?!
```

Comment:
we can remove this stuff: https://github.com/elastic/elasticsearch/pull/12966 it's deprecated since 2.0, no reason to keep the deprecated version of it.